core/src/main/scala/filodb.core/TargetSchemaProvider.scala

```diff
@@ -0,0 +1,11 @@
+package filodb.core
+
+import filodb.core.query.ColumnFilter
+
+trait TargetSchemaProvider {
+  def targetSchemaFunc(filter: Seq[ColumnFilter]): Seq[String]
```

Comment:
Target schema can change over time. The response should have a time dimension

Reply:
updated the code to consider target-schema changes over time.